src/librustc/ty/sty.rs

```diff
@@ -1814,20 +1814,28 @@ impl<'tcx> TyS<'tcx> {
 
     pub fn simd_type(&self, tcx: TyCtxt<'tcx>) -> Ty<'tcx> {
         match self.kind {
-            Adt(def, substs) => {
-                def.non_enum_variant().fields[0].ty(tcx, substs)
-            }
+            Adt(def, substs) => def.non_enum_variant().fields[0].ty(tcx, substs),
             _ => bug!("simd_type called on invalid type")
         }
     }
 
-    pub fn simd_size(&self, _cx: TyCtxt<'_>) -> usize {
+    pub fn simd_size(&self) -> usize {
```

Comment:
I think @gnzlbg's PR to allow `#[repr(simd)] struct Simd<T, const N: usize>([T; N]);` needs the `tcx` param here.

Reply:
Also, could this return `u64` instead, to match the `ty::layout` code?

Reply:
> I think @gnzlbg's PR to allow #[repr(simd)] struct Simd<T, const N: usize>([T; N]); needs the tcx param here.

Oh, I see. Such things should be explained in comments...

> Also, could this return u64 instead, to match the ty::layout code?

All the code in codegen is expecting a `usize` though.

Reply:
So I changed it to `u64` but had to insert a bunch of casts.

Reply:
The use of `usize` is likely ancient code not being updated to avoid `usize`, I think, so the casts can likely be avoided by enough strategical changes.

Reply:
`simd_size` is casting now, though. :/

Reply:
Yeah but it's a lossless cast, and it's in one place. Also, ideally everything other than `ty::layout` would look at the layout (again I think @gnzlbg does some of this in that one PR).